miniflux: 2.2.13 -> 2.2.14

ameertaweel opened the pull request:
https://miniflux.app/releases/2.2.14.html
https://github.com/miniflux/v2/releases/tag/2.2.14
diff: https://github.com/miniflux/v2/compare/v2.2.13...2.2.14

## Tests Failing Due to AppArmor

`nixosTests.miniflux` is currently failing because AppArmor on master.
The failure itself is unrelated to this `miniflux` bump.
See https://github.com/NixOS/nixpkgs/pull/443744#issuecomment-3303395290.

With the following line commented out of the VM test, things pass as intended:
`machine.fail('journalctl -b --no-pager --grep "^audit: .*apparmor=\\"DENIED\\""')`

## Things Done

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests] **(Failing only due to AppArmor)**.
  - [x] [Package tests] at `passthru.tests` **(Failing only due to AppArmor)**.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

emilylange commented:
> Unfortunately, I don't have a darwin machine to debug this. How can I go about this?

You should ignore this `nixpkgs-review` (`-gha`) spam.

`miniflux` does build on `-darwin`, just not with the sandbox enabled, which is no big deal. And, more importantly, this isn't a sudden regression caused by your version bump. Instead, this has been the case for quite some time now.

ameertaweel commented:
@emilylange thanks for the clarification.

emilylange commented on nixos/tests/miniflux.nix:
```diff
@@ -81,7 +81,7 @@ in
         systemd.services.postgresql-setup.postStart = lib.mkAfter ''
-          psql -tAd miniflux -c 'CREATE EXTENSION hstore;'
+          psql -tAd miniflux -c 'DROP EXTENSION IF EXISTS hstore;'
         '';
```
This VM test has no state. The `IF EXISTS hstore` will never be `true`, because you replaced the `CREATE EXTENSION hstore` with it.

You can just drop this altogether:

```suggestion
```

emilylange commented on nixos/modules/services/web-apps/miniflux.nix:
```diff
@@ -22,7 +22,7 @@ let
   pgbin = "${config.services.postgresql.package}/bin";
   preStart = pkgs.writeScript "miniflux-pre-start" ''
```
How about something like this:

```suggestion
  # The hstore extension is no longer needed as of v2.2.14
  # and would prevent Miniflux from starting.
  preStart = pkgs.writeScript "miniflux-pre-start" ''
```

ameertaweel replied:
Added comment.

emilylange commented on nixos/modules/services/web-apps/miniflux.nix:
```suggestion
         Whether a PostgreSQL database should be automatically created and
         configured on the local host.
```

A release note entry in the breaking change section for the few users that use `nixos/miniflux` with an external database might be a good idea too.

emilylange replied:
Wait no, you forgot to remove "and make sure to create the hstore extension in it" from the option description. Remove that, please. Looks like the change suggestion did not render the way I expected it to.

At which point you are amending your commit anyway, so please make your commit message include the link to the release notes I was referring to in <https://github.com/NixOS/nixpkgs/pull/454793#pullrequestreview-3372962226>.

emilylange reviewed (approved):
You used two different emails for the git author and git committer of your commit, one of which GitHub does not know. This is likely a configuration error with your git, but I figured I should tell you.

You can see this by running `git log --pretty=fuller` instead of `git log` locally. This is why GitHub renders your commit like the committer and author are two different persons.
You can also opt to add and verify your other email to your GitHub account. That won't fix your local git configuration, but it might fix how GitHub is currently rendering it.

Up to you to decide whether you consider this an issue to fix or not.

But if you do end up fixing your local git configuration, consider amending your commit to include the release notes in the commit message too, just like you did in your opening PR comment. That would be great. I forgot to mention it in my first review, sorry. Not blocking.

Either way, thanks!

ameertaweel committed:
miniflux: 2.2.13 -> 2.2.14

https://miniflux.app/releases/2.2.14.html
https://github.com/miniflux/v2/releases/tag/2.2.14
diff: https://github.com/miniflux/v2/compare/v2.2.13...2.2.14

ameertaweel commented:
My bad for forgetting to remove hstore from the service documentation.
I fixed it, fixed the two-email issue, and added release notes links in the commit message.

Thanks for your help!

emilylange reviewed (approved):
Awesome, thanks :)
